third_party/move/move-compiler-v2/src/pipeline/ability_processor.rs

```diff
@@ -99,6 +99,29 @@ impl FunctionTargetProcessor for AbilityProcessor {
             lifetime,
             exit_state,
         };
+
+        // Check whether unused parameters may be dropped
```

Comment:
This is the wrong way to fix this. Obviously parameters are not correctly treated by the drop analysis. Instead of adding some new ad-hoc code here which works around all the existing logic, please try to fix the real bug.

Reply:
Hi @wrwg, I think the issue happens because the drop information for parameters is missing in `CopyDropState` and the error is detected without analyzing the code. Do you think it makes more sense to extend `CopyDropState` with this info and add the check to the Transformer?

Reply:
I moved the check into the Transformer, which will be called before analyzing the code.